### PR TITLE
Consolidated functionality; 2 options to be tested

### PR DESCRIPTION
**Pull request by MrSquiggles13**
I commented out a line of code that could replace a whole block but have no way of testing either methods.

### CODE REVIEW

File: dags/blocksec_plugin/contract_interaction_operator.py [hunk 1]
```diff
@@ -69,9 +70,36 @@ def execute(self, context):
...
+
+        try:
+            receipt = self.web3.eth.get_transaction_receipt(transaction_hash)
+            confirmations = self.web3.eth.blockNumber - receipt.blockNumber
+        except TypeError: # Transaction has no block number
+            confirmations = 0
+        except TransactionNotFound:
+            confirmations = 0
+        print("Transaction {0} has {1} confirmations".format(transaction_hash, confirmations))
+        if confirmations >= self.confirmations:
```
Comment (mikeghen) [inline dags/blocksec_plugin/contract_interaction_operator.py]: self.confirmations needs to be added in the `init` method

File: dags/blocksec_plugin/contract_interaction_operator.py [hunk 2]
```diff
@@ -10,7 +11,7 @@ class ContractInteractionOperator(BaseOperator):
     Executes a generalized contract interaction
     """
 
-    template_fields = ['contract_address', 'function', 'args', 'abi_json']
+    template_fields = ['contract_address', 'function', 'args', 'abi_json', 'transaction_hash']
 
     @apply_defaults
     def __init__(self,
```
Comment (mikeghen) [inline dags/blocksec_plugin/contract_interaction_operator.py:17]: add `confirmations=3` as the default, add `self.confirmations = confirmations` into the init method like is done in the EthereumTransactionSensor's `__init__` method

**Review by mikeghen — CHANGES_REQUESTED** [review]

**Comment by shreyaspapi** [comment]
Tested works fine, safe to merge after #37

**Comment by shreyaspapi** [comment]
Tested works fine, safe to merge after #37

**Review by shreyaspapi — APPROVED** [review]

**Comment by MrSquiggles13** [comment]
Fixed logic of the execute method should be good to go

**Comment by MrSquiggles13** [comment]
@mikeghen mind helpin me out bud? I just wanna claim this bounty and be done with it im pretty sure its sound

**Pull request closed without merging**